agimus_controller/agimus_controller/ocp_base_croco.py

```diff
@@ -0,0 +1,226 @@
+                )
+                # Create Integrated Action Model (IAM), i.e. Euler integration of continuous dynamics and cost
+                runningModel = crocoddyl.IntegratedActionModelEuler(
+                    self._running_DAM, self.dt
+                )
+                runningModel.differential.armature = OCPParamsCrocoBase.armature
+                self._runningModelList.append(runningModel)
+            else:
+                terminalCostModel = crocoddyl.CostModelSum(self._state)
+                terminalCostModel.addCost(
+                    "stateReg",
+                    xRegCost,
+                    np.concatenate(
+                        [
+                            self._OCPParams.WeightedTrajectoryPoints[
+                                t
+                            ].weight.w_robot_configuration,
+                            self._OCPParams.WeightedTrajectoryPoints[
+                                t
+                            ].weight.w_robot_velocity,
+                        ]
+                    ),
+                )
+                terminalCostModel.addCost(
+                    "gripperPose",
+                    goalTrackingCost,
+                    self._OCPParams.WeightedTrajectoryPoints[
+                        t
+                    ].weight.w_end_effector_poses,
+                )
+
+                self._terminal_DAM = crocoddyl.DifferentialActionModelFreeFwdDynamics(
+                    self._state, self._actuation, terminalCostModel
+                )
+
+                self._terminalModel = crocoddyl.IntegratedActionModelEuler(
+                    self._terminal_DAM, 0.0
+                )
+                self._terminalModel.differential.armature = OCPParamsCrocoBase.armature
+
+        problem = crocoddyl.ShootingProblem(
+            self.x0, self._runningModelList, self._terminalModel
+        )
+        # Create solver + callbacks
+        ocp = mim_solvers.SolverSQP(problem)
+
+        # Merit function
+        ocp.use_filter_line_search = False
+
+        # Parameters of the solver
+        ocp.termination_tolerance = OCPParamsCrocoBase.termination_tolerance
+        ocp.max_qp_iters = OCPParamsCrocoBase.qp_iters
+        ocp.eps_abs = OCPParamsCrocoBase.eps_abs
+        ocp.eps_rel = OCPParamsCrocoBase.eps_rel
+
+        ocp.with_callbacks = OCPParamsCrocoBase.callbacks
+
+        result = ocp.solve(self.x_init, self.u_init, OCPParamsCrocoBase.solver_iters)
+
+        self.ocp_results = OCPResults(
+            states=ocp.xs,
+            ricatti_gains=ocp.Ks,  # Not sure about this one
+            feed_forward_terms=ocp.us,
+        )
+
+        return result
+
+    @abstractmethod
+    @property
+    def ocp_results(self) -> OCPResults:
+        """Output data structure of the OCP.
+
+        Returns:
+            OCPResults: Output data structure of the OCP. It contains the states, Ricatti gains and feed-forward terms.
+        """
+        return self._ocp_results
```

Comment:
There was a typo where you missed `_` and never populated this object actually

Reply:
good catch